src/components/ChannelSearch/ChannelSearch.tsx

```diff
@@ -136,7 +170,30 @@ const UnMemoizedChannelSearch = <
         { limit: 8, ...searchQueryParams?.options },
       );
 
-      setResults(users);
+      if (searchForChannels) {
+        const channelResponse = client.queryChannels(
+          // @ts-expect-error
+          {
+            name: { $autocomplete: text },
+            ...searchQueryParams?.filters,
+          },
+          {},
+          { limit: 5, ...searchQueryParams?.filters },
+        );
+
+        const [channels, { users }] = await Promise.all([channelResponse, userResponse]);
+
+        if (!channels && !users) setResults([]);
```

Comment:
pretty sure the api will return an empty array so always truthy, so change to !channels.length and !users.length

Reply:
actually seeing the next line where we spread, can just just delete the falsy check since we can spread the empty arrays